mmaction/datasets/pipelines/loading.py

```diff
@@ -910,6 +910,122 @@ def __call__(self, results):
 
 
 @PIPELINES.register_module()
+class AudioDecodeInit(object):
+    """Using librosa to initialize the audio reader.
+
+    Args:
+        io_backend (str): io backend where frames are store.
+            Default: 'disk'.
+        sample_rate (int): Audio sampling times per second. Default: 16000.
+
+    Required keys are "audio_path", added or modified keys are "length",
+    "sample_rate", "audios".
+    """
+
+    def __init__(self, io_backend='disk', sample_rate=16000, **kwargs):
+        self.io_backend = io_backend
+        self.sample_rate = sample_rate
+
+        self.kwargs = kwargs
+        self.file_client = None
+
+    def __call__(self, results):
+        """Perform the librosa initialization.
+
+        Args:
+            results (dict): The resulting dict to be modified and passed
+                to the next transform in pipeline.
+        """
+        try:
+            import librosa
+        except ImportError:
+            raise ImportError('Please install librosa first.')
+
+        if self.file_client is None:
+            self.file_client = FileClient(self.io_backend, **self.kwargs)
+        try:
+            file_obj = io.BytesIO(self.file_client.get(results['audio_path']))
+            y, sr = librosa.load(file_obj, sr=self.sample_rate)
+        except BaseException:
+            # Generate a ramdom dummy 10s input
+            y = np.random.randn(10.0 * self.sample_rate, )
+            sr = self.sample_rate
+
+        results['length'] = y.shape[0]
+        results['sample_rate'] = sr
+        results['audios'] = y
+        return results
+
+
+@PIPELINES.register_module()
+class LoadAudioFeature(object):
+    """Load offline extracted audio features.
+
+    Required keys are "audio_path", added or modified keys are "length",
+    audios".
+    """
+
+    def __init__(self):
+        pass
+
+    def __call__(self, results):
+        """Perform the numpy loading.
+
+        Args:
+            results (dict): The resulting dict to be modified and passed
+                to the next transform in pipeline.
+        """
+        try:
+            feature_map = np.load(results['audio_path'])
+        except BaseException:
+            # Generate a ramdom dummy 10s input
```

Comment:
random

Reply:
why is a random signal better than an all-zero one?

Reply:
Actually I don't know. Around 1/20 videos in Kinetics400 do not have audio stream, other papers like AvSlowfast never mentioned this issue.

I think adding random samples in training may lead to more robust optimization, many all-zero samples can cause overfitting to this 'shortcut', especially when a certain action like "dumb show"(just for an example) have many muted videos.

Reply:
ramdom is a typo here

Reply:
exception is expensive https://docs.python.org/3/faq/design.html#how-fast-are-exceptions

if you can use `exists()` etc. to avoid exceptions, you can also cache the existence in memory to avoid extra disk access

Reply:
Exceptions changed to if branches.